### PR TITLE
Part 4 redo

### PR DESCRIPTION
**Pull request by belv2c**

### CODE REVIEW

File: javascripts/controllers/FavoritesCtrl.js [hunk 1]
```diff
@@ -1,5 +1,53 @@
...
+$scope.favoriteContacts = (contact) => {
+	let updatedContact = {};
+
+	if (!contact.favorite) {
+		updatedContact = ContactService.createContactObject(contact);
+	} else {
+		updatedContact = ContactService.createContactObject(contact);
+		updatedContact.favorite = false;
+	}
+	ContactService.updateContact(updatedContact, contact.id).then(() => {
```
Comment (morecallan) [inline javascripts/controllers/FavoritesCtrl.js:35]: There's a weird bug here... when I click on the favorite button, the first and last name, birthday and nickname all disappear!

File: javascripts/controllers/ViewCtrl.js [hunk 2]
```diff
@@ -20,4 +20,28 @@ app.controller("ViewCtrl", function($rootScope, $scope, ContactService) {
...
+$scope.favoriteContacts = (contact) => {
+	let updatedContact = {};
+
+	if (!contact.favorite) {
+		updatedContact = ContactService.createContactObject(contact);
+	} else {
+		updatedContact = ContactService.createContactObject(contact);
+		updatedContact.favorite = false;
+	}
+	ContactService.updateContact(updatedContact, contact.id).then(() => {
```
Comment (morecallan) [inline javascripts/controllers/ViewCtrl.js:32]: See the other comment about weird bug happening here.

File: javascripts/controllers/EditCtrl.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+app.controller("EditCtrl", function($location, $rootScope, $routeParams, $scope, ContactService) {
+	
+const getContacts = () => {
+		ContactService.getOneContact($routeParams.id).then((results) => {
+			$scope.contact = results.data;
```
Comment (morecallan) [inline javascripts/controllers/EditCtrl.js:7]: This line isn't actually working. It should be loading the contact information you are pulling back here into the fields in the edit partial.

**Review by morecallan — COMMENTED** [review]
Hey @belv2c - just a few things to look over here!

File: javascripts/controllers/FavoritesCtrl.js (hunk 1 above)
Comment (belv2c) [reply javascripts/controllers/FavoritesCtrl.js:35]: Fixed this bug

File: javascripts/controllers/ViewCtrl.js (hunk 2 above)
Comment (belv2c) [reply javascripts/controllers/ViewCtrl.js:32]: Fixed this bug

File: javascripts/controllers/EditCtrl.js (hunk 3 above)
Comment (belv2c) [reply javascripts/controllers/EditCtrl.js:7]: Edit fields have contact info now and update the contact

File: javascripts/controllers/FavoritesCtrl.js [hunk 4]
```diff
@@ -1,5 +1,53 @@
...
+
+$scope.favoriteContacts = (contact) => {
+	let updatedContact = {};
+
+	if (!contact.favorite) {
+		updatedContact = ContactService.createContactObject(contact);
+	} else {
+		updatedContact = ContactService.createContactObject(contact);
+		updatedContact.favorite = false;
+	}
```
Comment (morecallan) [inline javascripts/controllers/FavoritesCtrl.js:34]: So this  "update favorite" functionality isn't working for the favorites page...

**Review by morecallan — COMMENTED** [review]
@belv2c - nice work! One small bug. Definitely needs some debugging to find this one.

File: javascripts/controllers/FavoritesCtrl.js [hunk 5]
```diff
@@ -1,5 +1,36 @@
...
+
+$scope.deleteContact = (contactId) => {
+		ContactService.deleteContact(contactId).then((results) => {
+			getContacts();
+		}).catch((err) => {
+			console.log("error in deleteContact", err);
+		});
+	};
+
+
```
Comment (morecallan) [inline javascripts/controllers/FavoritesCtrl.js]: @belv2c - Oh! It's because it got removed from here!

**Comment by morecallan** [comment]
👉 poke @belv2c

**Pull request merged**